src/CoreBundle/Bridge/Chameleon/Module/Sidebar/SidebarBackendModule.php

```diff
@@ -160,25 +205,88 @@ private function getMenuItems(): array
             }
         }
 
+        $popularCategory = $this->getPopularMenuEntries($menuItemMap);
+        if (null !== $popularCategory) {
+            \array_unshift($menuCategories, $popularCategory);
+        }
+
         return $menuCategories;
     }
 
-    private function getActiveCategory(): ?string
+    private function getOpenCategories(): array
+    {
+        $session = $this->getSession();
+
+        if (null === $session) {
+            return [];
+        }
+
+        return $session->get(self::OPEN_CATEGORIES_SESSION_KEY, []);
+    }
+
+    private function getPopularMenuEntries(array $menuItemMap): ?MenuCategory
     {
-        $session = $this->requestStack->getCurrentRequest()->getSession();
+        $activeUser = \TCMSUser::GetActiveUser();
+        if (null === $activeUser) {
+            return null;
+        }
 
-        return $session->get(self::ACTIVE_CATEGORY_SESSION_KEY);
+        $menuItemsClickedByUser = $this->userMenuItemDataAccess->getMenuItemIds($activeUser->id);
+        $menuItemsClickedByUser = \array_slice($menuItemsClickedByUser, 0, 6);
+
+        $items = [];
+
+        foreach ($menuItemsClickedByUser as $menuId) {
+            if (\array_key_exists($menuId, $menuItemMap)) {
+                $items[] = $menuItemMap[$menuId];
+            }
+        }
+
+        if (0 === $items) {
+            return null;
+        }
+
+        return new MenuCategory(
+            self::POPULAR_CATEGORY_ID,
+            $this->translator->trans('chameleon_system_core.sidebar.popular_entries'),
+            'fas fa-fire',
+            $items
+        );
     }
 
+    /**
+     * @deprecated since 6.3.8 - use toggleCategoryOpenState
+     */
     protected function saveActiveCategory(): void
     {
-        $activeCategory = $this->inputFilterUtil->getFilteredPostInput('categoryId');
-        if ('' === $activeCategory) {
-            $activeCategory = null;
+        $this->toggleCategoryOpenState();
+    }
+
+    protected function toggleCategoryOpenState(): void
+    {
+        $session = $this->getSession();
+        if (null === $session) {
+            return;
         }
 
-        $session = $this->requestStack->getCurrentRequest()->getSession();
-        $session->set(self::ACTIVE_CATEGORY_SESSION_KEY, $activeCategory);
+        $toggledCategoryId = $this->inputFilterUtil->getFilteredPostInput('categoryId', '');
+
+        if ('' === $toggledCategoryId) {
+            return;
+        }
+
+        $activeCategoryIds = $session->get(self::OPEN_CATEGORIES_SESSION_KEY, []);
+
+        $index = array_search($toggledCategoryId, $activeCategoryIds, true);
+        if (false !== $index){
+            unset($activeCategoryIds[$index]);
+        } else {
+            $activeCategoryIds[] = $toggledCategoryId;
+        }
+
+        $session->set(self::OPEN_CATEGORIES_SESSION_KEY, $activeCategoryIds);
+
+        // TODO could/should save this connected to the user (and not login session)?
```

Comment:
I would reset this on login and not save it in the user object. The current implementation is o.k. for me.

Reply:
Comment removed